pkg/cloud/amazon/s3_storage.go

```diff
@@ -617,7 +617,7 @@ func (s *s3Storage) newClient(ctx context.Context) (s3Client, string, error) {
 		})
 
 		creds := stscreds.NewAssumeRoleProvider(client, s.opts.assumeRoleProvider.roleARN, withExternalID(s.opts.assumeRoleProvider.externalID))
-		cfg.Credentials = creds
+		cfg.Credentials = aws.NewCredentialsCache(creds)
```

Comment:
Could we add in the commit description/in a comment the reason for this change for future reference? A link to the AWS documentation is sufficient: https://docs.aws.amazon.com/sdk-for-go/v2/developer-guide/configure-gosdk.html#specify-credentials-programmatically.